File: tests/nodejs/cases/case_097__session.js
/**
 * Task case 097: _session
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case097Session(input) {
  // Task: internal session records and prepare transport-ready payload.
  const taskName = '_session';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'session',
    stage: '',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is appended to retained audit state, outliving function scope.
  retainedAudit.push(payload);
  return 'ok';
}

module.exports = {
  case097Session,
};

File: tests/python/cases/case_097__session.py
"""
Task case 097:  session payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_097__session(input_data):
    # Task:  session records and prepare transport-ready payload.
    task_name = '_session'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'session',
        'stage': '',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is appended to RETAINED_AUDIT, outliving function scope.
    RETAINED_AUDIT.append(payload)
    return 'ok'
